src/erasure-code/ErasureCodePlugin.cc

```diff
@@ -39,15 +39,13 @@ ErasureCodePluginRegistry::ErasureCodePluginRegistry() = default;
 
 ErasureCodePluginRegistry::~ErasureCodePluginRegistry()
 {
-  if (disable_dlclose)
-    return;
-
   for (std::map<std::string,ErasureCodePlugin*>::iterator i = plugins.begin();
-       i != plugins.end();
-       ++i) {
+       i != plugins.end();) {
     void *library = i->second->library;
     delete i->second;
-    dlclose(library);
+    i = plugins.erase(i);
+    if (!disable_dlclose)
+      dlclose(library);
```

Comment:
can we just put something like
```c++
#include <ranges>
//..
for (auto const plugin : plugins | std::ranges::views::values) {
  void *library = plugin->library;
  delete plugin;
  if (disable_dlclose) {
    dlclose(library);
  }
}
```
at least we don't need to use `i = plugins.erase(i);`

Reply:
Make Sense, adjusted.

Reply:
Seems like clang14 doesn't work well with views. Ref: https://stackoverflow.com/questions/73929080/error-with-clang-15-and-c20-stdviewsfilter

Reply:
Done